```
.
├── Load_PotReal.R
├── Load_segDrift.R
├── README.md
├── ROH_tryruns.R
├── ROHs.R
├── Rxy.R
├── admixture.R
├── analyzeSLiMOutput.R
├── diversity.R
├── getSLiMoutput.R
├── heterozygosity_norel.R
├── mut_age.R
├── mutation_age.R
├── population_tree.R
├── recomb.R
└── simple-sensitivity-analysis-with-r
    ├── jump-data.csv
    └── sensitivity-analysis.R
```

#### Details ####

1. _Adapter removal_ : Trim raw sequences to remove adapters and low-quality bases (Phred < 20) using Trimmomatic
2. _Alignment.sh_ : Mapping and read processing following GATK “Best Practice Workflow”
3. _Base_recalibration.sh_ : Recaliberate mapped sequences using known variants from high quality sequnce reads
4. _Depth&Breadth.sh_ : Calculate mapping stats using SAMtools

##### VariantCalling #####

1. _HaplotypeCaller.sh_ : Call genome-wide variants in each sample (GVCF) using GATK v4
2. _GenotypeGVCFs.sh_: Genotype calling from combined GVCF information of all samples and SNP filtering
3. _vcf_filtering.sh_ : SNP filtering using VCFtools

##### Diversity #####

1. _vcfstats.sh_ : Calculate diversity, Fst, and other SNP statistics using VCFtools
2. _HeterozygosityKb.sh_ : Divide genome-wide SNPs by chromosomes and calucate heterozygosity in 1 Mb windows
3. _norelMAF.sh_ : Comparing diversity estimates between Arizona and Texas populations after removing close relatives
4. _tassel.sh_ : Create phylogenetic trees using genome-wide SNP data

##### ROH,Ancestry,&Demography #####

1. _ROH_analysis.sh_ : Estimating ROHs using parameters estimated from a sensitivity analysis 
2. _HaplotypePhasing.sh_ : Phasing genotypes into haplotypes via Beagle using chicken recombination maps
3. _finestructure.sh_ : Pairwise Co-ancestry estimation from shared IBD segments using fineStructure
4. _gadma_3pop.sh_ : Demographic modeling for 3-population model (AZ-TX-MX) using GADMA
5. _gadma_2pop.sh_ : Demographic modeling for 2-population model (AZ-TX) using GADMA
6. _gadma_boostrap.sh_ : Bootstrapping final demographic model from 100 SFS generated by bootstrapping samples


##### MutationAge&Load #####

1. _snp_annotation.sh_ : Functional annotation of SNPs and deleteriousness estimation (SIFT) using VEP
2. _genetic_load.sh_ : Genotype and minor allele frequency (MAF) extraction of each individual and SNPs from the VCF file
3. _mutation_age.sh_ : Mutation age estimation of deleterious mutations from phased haplotypes using GEVA

##### SLiM_Simulations #####

1. _1k.prebottle.slim_ : SLiM script for evolutionary genomic simulations (Time Epoch 1: Burn-in pre-expasion)
2. _1k.expand.slim_ : SLiM script for evolutionary genomic simulations (Time Epoch 2: Ancestal expansion and founding bottleneck)
3. _1k.postbottle.slim_ : SLiM script for evolutionary genomic simulations (Time Epoch 3: Post-bottleneck until present)
4. _getSLiMoutput.sh_ : Summarize results from 100 independent runs of SLiM simutations
5. _slimFst_ : Calculate Fst between two populations over time after founding bottleneck
